### PR TITLE
`dart pub remove`: Remove top-level key if empty after operation.

### PR DESCRIPTION
**Pull request by raunaqarora21**
The changes made to this code improve the way dependencies are removed from the pubspec.yaml file. The previous version of the code only removed the package from the dependencies or dev_dependencies key, but it didn't check if the key was now empty after the removal. The updated version now also checks if the key is empty after the removal, and if it is empty, it removes the key as well. This ensures that the pubspec.yaml file is kept clean and organized.

Fixes https://github.com/dart-lang/pub/issues/3530

### CODE REVIEW

**Comment by sigurdm** [comment]
Hi - thanks for working on this.

Could you try adding a test for this new behaviour.

The test should go in: test/remove/remove_test.dart

**Comment by raunaqarora21** [comment]
Hi @sigurdm 
I have added the test case in: test/remove/remove_test.dart
But it is not working as expected, so can you please review and suggest the changes if required

File: lib/src/command/remove.dart [hunk 1]
```diff
@@ -143,13 +143,13 @@ class RemoveCommand extends PubCommand {
 
         if (dependenciesNode is YamlMap &&
             dependenciesNode.containsKey(package)) {
-          if (dependenciesNode.length == 1) {
+          yamlEditor.remove([dependencyKey, package]);
+          found = true;
+          // Check if the dependencies or dev_dependencies map is now empty
+          // If it is empty, remove the key as well
+          if (dependenciesNode.isEmpty) {
```
Comment (sigurdm) [inline lib/src/command/remove.dart:150]: I don't think the `dependenciesNode` gets mutated by the `yamlEditor.remove([dependencyKey, package]);` above.

You could probably replace this with something like:
```suggestion
          if ((yamlEditor.parseAt([dependencyKey]) as YamlMap).isEmpty) {
```

**Comment by raunaqarora21** [comment]
@sigurdm - Thanks for the suggestion. 
Now the test case is getting passed

**Review by sigurdm — APPROVED** [review]
LGTM

**Comment by sigurdm** [comment]
Thanks for the contribution!

**Pull request merged**